modal/image.py

```diff
@@ -831,11 +831,14 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
         return _Image._from_args(
             base_images={"base": self},
             dockerfile_function=build_dockerfile,
-            context_mount_function=lambda: _Mount._from_local_file(local_path, remote_path=f"/{basename}"),
+            context_mount_function=lambda: _Mount._from_local_file(local_path, remote_path=f"/{basename}"),  # pyright: ignore [reportGeneralTypeIssues=false]
         )
 
     def add_local_python_source(
-        self, *modules: str, copy: bool = False, ignore: Union[Sequence[str], Callable[[Path], bool]] = NON_PYTHON_FILES
+        self,
+        *modules: str,
+        copy: bool = False,
+        ignore: Optional[Union[Sequence[str], Callable[[Path], bool]]] = NON_PYTHON_FILES,
```

Comment:
added `Optional` here since that's what `_from_local_python_packages()`'s `ignore` param seems to expect later.